pkg/shell/index.js

```diff
@@ -336,14 +362,20 @@ define([
         }
 
         if (address) {
+            machine = machines.lookup(address);
 
             /* If the machine is not available, then redirect to dashboard */
-            machine = machines.lookup(address);
-            if (!machine) {
-                if (listing)
+            if (!machine || machine.problem) {
+                if (!machine && !machines.read_complete) {
+                    show_connecting();
+                } else if (machine && reconnect && machine.problem) {
```

Comment:
It's not really about the machine but about if the navigation call is triggered by an action or triggered by one of the machine event callbacks. Machine is shared around, and this information is very local to what index is doing with navigation.